app/services/testing_services/process_tested_submission.rb

```diff
@@ -18,6 +18,22 @@ def call
         results: results
       )
       submission.update!(tested: true)
+
+      status = if submission.tested?
+                 submission.test_results.last.results_status
+               else
+                 :failed_dependency
+               end
+
+      message = submission.test_results.last.message if status == :error
+      failure = submission.test_results.last.tests.select{|t|t['status'] == 'fail'}.first if status == :fail
+
+      SubmissionChannel.broadcast_to(
```

Comment:
does `broadcast_to` mean that all connected browsers will receive the message?

Reply:
I assume this works across process  - assuming that service is invoked via a rake-based propono listener, then it must be. Does that mean that the plumbing is all handled by redis? I assume that means we need a single redis for both webapps and processor to share in order for delivery to work to any connected consumer.

Reply:
If we have a single redis for all of this, do we want to deploy it standalone (so that it's an isolated single point of failure) from everything else?

Reply:
It would be useful to understand what would happen if redis were to go rogue and we couldn't deliver messages. Would the web UI continue to work (albeit with the browser needing the occassional refresh). I guess I'm asking whether websockets are an enhancement or a critical component, because that informs decisions about resilience of the redis broker

Reply:
Do we have any idea how transient the subscription-data in redis is? I imagine it's pretty transient but it would be good to confirm.

Reply:
I would probably say that websockets are an enhancement in order to deliver the data without a page refresh. I think we should still save everything to the database so that when the user refreshes they would get the same page as with when websockets delivers the data.

I'm not sure how long the timeout is before Rails cleans up subscription information. The only thing I know is that each subscription is an open pub sub channel between redis and the server and would be maintained the moment we'd need more messages.

Again, I think these questions are also a matter of how we would want the UI to be. If we aren't saving the build runs as what I've said above, we might have lots of long running connections between each user solving a problem versus a lot more short-lived connection for each user when modelling it as a separate channel for each "build". It would good to discuss what the pros and cons are for each and what user experience we are aiming for.